pkg/process/procutil/process_linux_test.go

```diff
@@ -204,3 +204,137 @@ func TestProcfsChange(t *testing.T) {
 	assert.NotContains(t, newProcByPID2, int32(29613))
 	assert.Contains(t, procByPID, int32(29613))
 }
+
+func TestParseStatusLine(t *testing.T) {
+	probe := NewProcessProbe()
+	defer probe.Close()
+
+	for _, tc := range []struct {
+		line     []byte
+		expected *statusInfo
+	}{
+		{
+			line: []byte("Name:\tpostgres"),
+			expected: &statusInfo{
+				name:        "postgres",
+				memInfo:     &MemoryInfoStat{},
+				ctxSwitches: &NumCtxSwitchesStat{},
+			},
+		},
+		{
+			line: []byte("Name:\t\t\t\t\tpostgres"),
+			expected: &statusInfo{
+				name:        "postgres",
+				memInfo:     &MemoryInfoStat{},
+				ctxSwitches: &NumCtxSwitchesStat{},
+			},
+		},
+		{
+			line: []byte("State:\tS (sleeping)"),
+			expected: &statusInfo{
+				status:      "S",
+				memInfo:     &MemoryInfoStat{},
+				ctxSwitches: &NumCtxSwitchesStat{},
+			},
+		},
+		{
+			line: []byte("Uid:\t112\t112\t112\t112"),
+			expected: &statusInfo{
+				uids:        []int32{112, 112, 112, 112},
+				memInfo:     &MemoryInfoStat{},
+				ctxSwitches: &NumCtxSwitchesStat{},
+			},
+		},
+		{
+			line: []byte("bad status line"), // bad status
+			expected: &statusInfo{
+				memInfo:     &MemoryInfoStat{},
+				ctxSwitches: &NumCtxSwitchesStat{},
+			},
+		},
+		{
+			line: []byte("Name:\t"), // edge case for parsing failure
+			expected: &statusInfo{
+				memInfo:     &MemoryInfoStat{},
+				ctxSwitches: &NumCtxSwitchesStat{},
+			},
+		},
+		{
+			line: []byte("NSpid:\t123"),
+			expected: &statusInfo{
+				nspid:       123,
+				memInfo:     &MemoryInfoStat{},
+				ctxSwitches: &NumCtxSwitchesStat{},
+			},
+		},
+		{
+			line: []byte("VmRSS:\t712 kB"),
+			expected: &statusInfo{
+				memInfo:     &MemoryInfoStat{RSS: 712 * 1024},
+				ctxSwitches: &NumCtxSwitchesStat{},
+			},
+		},
```

Comment:
Should we add cases for `Gid`, `Threads`, `voluntary_ctxt_switches`, `nonvoluntary_ctxt_switches`, `VmSize` and `VmSwap` just in case we accidentally break the parse logic for any of these?

Reply:
Also a case for `Name` with spaces and tabs?

Reply:
I added one(or more) test cases for each field.